### DISCUSSION TITLE
set HasHeaders to true by default

### DISCUSSION THREAD

**Discussion by pbelouin-mpiwg** [jgile/vue-csv-import#92] [Q&A]
Hello, I would like to set up the components so that it is implied that the uploaded CSV has headers - not sure exactly how to do this, I thought doing:

```
    <vue-csv-import
      v-model="csv"
      :has-headers="True"
      :fields="{
        name: { required: false, label: 'Name' },
        location: { required: true, label: 'Location' },
      }"
    >
      <vue-csv-errors></vue-csv-errors>
      <vue-csv-input></vue-csv-input>
      <vue-csv-map></vue-csv-map>
    </vue-csv-import>
```

Would work, but this doesn't seem to do the trick!

Another question is, can I remove the map component, and just map the csv by default?

Thanks a lot in advance!

**Comment by James-Burgess** [comment]
Heyo, Thanks for trying out vue-csv-import!
As it stands now, there is no option to pass into the mapper, however, You can pass attrs the the checkbox component and set it to be checked by default:

```html
 <vue-csv-toggle-headers :checkboxAttributes="{checked: true}" />
```
You can then take this one step further and hide the checkbox with some css:
```html
<vue-csv-toggle-headers :checkboxAttributes="{checked: true}" :labelAttributes="{style: 'display: none'}"/>
```

For your other question about negating the mapper, that is not possible at this time, however the same trick of passing attrs is possible too. 

```html
<vue-csv-map style="display: none"/>

```

**Reply by pbelouin-mpiwg** [reply]
Hi! Thank you for the prompt answer! Unfortunately it seems that just checking the checkbox by default doesn't do the trick, I assume the value needs to be set up in another way, but I have to admin my vue is a bit rusty so not entirely sure how to do this just yet - in any case, thank you very much, I think I'm not too far now!
